src/Sarif/ExtensionMethods.cs

```diff
@@ -15,6 +15,37 @@ namespace Microsoft.CodeAnalysis.Sarif
 {
     public static class ExtensionMethods
     {
+        /// <summary>
+        /// Attempt to reconstruct a URI, if appropriate, using Run instance 
+        /// originalUriBaseId and uriBaseId properties. If this method cannot
+        /// successfully reconstitute an absolute URI, it will return false
+        /// and populate 'resolvedUri' with the input value.
+        /// </summary>
+        /// <param name="run"></param>
+        /// <param name="uri"></param>
+        /// <param name="resolvedUri"></param>
+        /// <returns></returns>
+        public static bool TryResolveUri(this Run run, FileLocation fileLocation, out Uri resolvedUri)
+        {
+            resolvedUri = fileLocation.Uri;
+
+            // We can't restore any absolute URIs unless someone has
+            // deconstructed them using uriBaseId + originalUriBaseIds
+            if (run.OriginalUriBaseIds == null) { return false; }
```

Comment:
>false [](start = 57, length = 5)

I'm pondering this. I think it's more common, when returning `false` from a `TryXxx` method, to set the `out` parameter to `null`. I can't deny the usefulness of populating it with the input URI, but I don't know if it violates "least surprise". Your call.

Reply:
Update: See my comment in InsertOptionalDataVisitor.cs. This _does_ violate least surprise, and I think you should change the initialization to `null`.

---
In reply to: [205979160](https://github.com/Microsoft/sarif-sdk/pull/974#discussion_r205979160) [](ancestors = 205979160)